### PR TITLE
Fix: grouped-label classification when use_labels_groups=True need to use logits =false.

### LINKED ISSUE #3804: SCANVI bug — classify_helper divides logits by their sum
- Classifier.__init__ default = logits=False 
  - SCANVAE explicitly overrides the default to logits=True at _scanvae.py:142 
  - Therefore SCANVAE's self.classifier(z) returns raw logits 
  - Therefore the classify_helper factoring divides raw logits (which can be negative) by their sum (lines 779-791 in _base_module.py) 
            @auto_move_data
              def classify_helper(self, z):
                  if self.use_labels_groups:
                      w_g = self.classifier_groups(z)
                      unw_y = self.classifier(z)
                      w_y = torch.zeros_like(unw_y)
                      for i, group_index in enumerate(self.groups_index):
                          unw_y_g = unw_y[:, group_index]
                          w_y[:, group_index] = unw_y_g / (unw_y_g.sum(dim=-1, keepdim=True) + 1e-8)
                          w_y[:, group_index] *= w_g[:, [I]]
                  else:
                      w_y = self.classifier(z)
                  return w_y

```python
# Your code here
```

```pytb
[Paste the error output produced by the above code here]
```

#### Versions:

1.4.2

> VERSION

### PR DESCRIPTION
**Pull request by ori-kron-wis**
close https://github.com/scverse/scvi-tools/issues/3804

### CODE REVIEW

File: src/scvi/module/_scanvae.py [hunk 1]
```diff
@@ -142,6 +142,9 @@ def __init__(
             "logits": True,
         }
         cls_parameters.update(classifier_parameters)
+        # group normalization in classify_helper requires probabilities, not raw logits
+        if use_labels_groups:
+            cls_parameters["logits"] = False
```
[the remark this answers is not in the document]
Comment (ori-kron-wis) [reply src/scvi/module/_scanvae.py]: @codex address this issue

[the remark this answers is not in the document]
Comment (ori-kron-wis) [reply src/scvi/module/_scanvae.py]: @codex address this issue
